c10/core/TensorImpl.h

```diff
@@ -602,9 +628,12 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return sym_strides_default();
   }
   inline c10::SymIntArrayRef sym_strides_default() const {
-    return c10::SymIntArrayRef(
-        reinterpret_cast<const c10::SymInt*>(sizes_and_strides_.strides_data()),
-        sizes_and_strides_.size());
+    if (has_symbolic_sizes_strides_) {
+      return extra_meta_->strides_;
+    } else {
+      // strides guaranteed to be non-negative, so unchecked cast is OK
+      return c10::SymIntArrayRef::fromIntArrayRefUnchecked(strides_default());
```

Comment:
this is an example use site where fromIntArrayRefKnownNonnegative would be useful (and we could add a TORCH_INTERNAL_ASSERT_DEBUG_ONLY that it actually is non-negative in that wrapper (or in fromIntArrayRefUnchecked in general if "known nonnegative" and "safe to be unchecked" are synonymous); in fact the more I think about it the more I like marking callsites that are unchecked precisely because they are known non-negative vs known not-symint because it will simplify potential representation changes